platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,122 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[no_source_in_root]]
+== Do Not Put Source Files in the Root Project
+:keywords: structuring-builds
+
+Do not put source files in your root project; instead, put them in a separate project.
+
+=== Explanation
+
+The root project is a special project in Gradle that serves as the entry point for your build.
+
+It is the place to configure settings that apply globally to the entire build.
```

Comment:
How does it align with isolated projects/convention plugins?

And from beginner view: what's the difference to a settings.build file?

Reply:
This design pattern should work fine with IP or convention plugins.

I think your second point has some merit.  Most settings are meant to go in `settings.gradle`.  I'll will add a sentence and link to docs clarifying this.